src/config.rs

```diff
@@ -0,0 +1,99 @@
+//! Configuration for `HttpClient`s.
+
+/// Configuration for `HttpClient`s.
+#[non_exhaustive]
+#[derive(Clone, Debug)]
+pub struct Config {
+    /// TCP `NO_DELAY`.
+    pub no_delay: Option<Option<bool>>,
+    /// Connection timeout duration.
+    pub timeout: Option<Option<u64>>,
+}
+
+impl Config {
+    /// Construct new empty config.
+    pub fn new() -> Self {
+        Self {
+            no_delay: None,
+            timeout: None,
+        }
+    }
+
+    /// Construct a new chainable configuration builder.
+    pub fn builder() -> ConfigBuilder {
+        ConfigBuilder::new()
+    }
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl From<ConfigBuilder> for Config {
+    fn from(builder: ConfigBuilder) -> Self {
+        builder.build()
+    }
+}
+
+/// A chainable builder for `Config`.
+#[derive(Clone, Debug)]
+pub struct ConfigBuilder(Config);
+
+impl ConfigBuilder {
```

Comment:
Why make ConfigBuilder different from Config and add Config::with_nodelay(self, bool) and Config::with_timeout(self, Duration), since they're already Options.  Usage would look like `H1Client::new().with_config(Config::new().with_timeout(Duration::from_secs(10)).with_nodelay(true))`